### DISCUSSION TITLE
Multiple use of a loaded voice.

### DISCUSSION THREAD

**Discussion by MartinDavidWaller** [rhasspy/piper#215] [General]
Hi,

Many thanks for doing this project, the voice quality is exceptional.

Loading the voice data seems to be a bit of a bottle neck. If, using the classic, echo 'hello' | piper .... type command I want to say multiple sentences then I can but each time the code must load the voice which, as you will know, can take a few seconds. 

For an application I have in mind I'd like to be able to write library layer on top of piper that pre-loads a voice and then allow multiple calls into the library to create many .wav files only having loaded the voice once.

I was wondering if you had any documentation that might point me in the right direction to write such a library.

Many thanks

Martin

**Comment by rhohndorf** [comment]
Assuming you are using Python you can just instance a voice object, which loads the model
```
model = "en_GB-semaine-medium.onnx"
voice = PiperVoice.load(model, ...)
```

and then make repeated calls to 
`voice.synthesize(...)`

**Reply by MartinDavidWaller** [reply]
Hi Ruben,

Many thanks for the reply. 

I'm not using Python. I'm hoping to call the libraries from C#. What you gave me has provided me with some clues how to proceed.

Many thanks

Martin

**Comment by StoryHack** [comment]
You can also use a jsonl file as input. It keeps the voice loaded while processing each line of json. See the section on the main page under "JSON Input"
